bokeh/models/annotations.py

```diff
@@ -111,8 +112,9 @@ class BoxAnnotation(Annotation):
 
     """
 
-    left = Either(Auto, NumberSpec(), default=None, help="""
-    The x-coordinates of the left edge of the box annotation.
+    left = NumberSpec(default=None, help="""
+    The x-coordinates of the left edge of the box annotation. If value is set
+    to `None`, the box annotation will continue to the edge of the plot area.
     """)
```

Comment:
Ok - I can't currently remember my reasoning for removing that. I'll add it back (it shouldn't impact my code)